benchmark/benchmark17_test.go

```diff
@@ -0,0 +1,85 @@
+// +build go1.7
+
+package benchmark
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/benchmark/stats"
+)
+
+func BenchClientUnary(b *testing.B, maxConcurrentCalls, reqSize, respSize int) {
+	grpc.EnableTracing = true
+	runUnary(b, maxConcurrentCalls, reqSize, respSize)
+}
+
+func BenchClientStream(b *testing.B, maxConcurrentCalls, reqSize, respSize int) {
+	grpc.EnableTracing = true
+	runStream(b, maxConcurrentCalls, reqSize, respSize)
+}
+
+func BenchClientUnaryNoTrace(b *testing.B, maxConcurrentCalls, reqSize, respSize int) {
+	grpc.EnableTracing = false
+	runUnary(b, maxConcurrentCalls, reqSize, respSize)
+}
+
+func BenchClientStreamNoTrace(b *testing.B, maxConcurrentCalls, reqSize, respSize int) {
+	grpc.EnableTracing = false
+	runStream(b, maxConcurrentCalls, reqSize, respSize)
+}
+
+func BenchmarkClient(b *testing.B) {
+	benchmarks := []struct {
+		maxConcurrentCalls int
+		reqSize            int
+		respSize           int
+	}{
+		{1, 1, 1},
+		{8, 1, 1},
+		{64, 1, 1},
+		{512, 1, 1},
+		{1, 1, 1024},
+		{1, 1024, 1},
+		{1, 1024, 1024},
+		{8, 1, 1024},
+		{8, 1024, 1},
+		{8, 1024, 1024},
+		{64, 1, 1024},
+		{64, 1024, 1},
+		{64, 1024, 1024},
+		{512, 1, 1024},
+		{512, 1024, 1},
+		{512, 1024, 1024},
+	}
+
+	for _, bm := range benchmarks {
+		maxC, reqS, respS := bm.maxConcurrentCalls, bm.reqSize, bm.respSize
+
+		b.Run(fmt.Sprintf("Unary-Trace maxConcurrentCalls: "+
```

Comment:
This line generates the bench name. Spaces are replaced with `_` automatically. So, the string generated is:
`Unary-Trace maxConcurrentCalls: 1, reqSize: 1, respSize: 1`
the result shows:
`Unary-Trace_maxConcurrentCalls:_1,_reqSize:_1,_respSize:_1-12`

How about remove the ` ` and `:` from `Sprintf`, and reformat it as:
`Unary-Trace-maxConcurrentCalls_%v-reqSize_%v-respSize_%v`,
result will be:
`Unary-Trace-maxConcurrentCalls_1-reqSize_1-respSize_1`